Devc 580 Create generic basic heat sensor DTH for FireAngel Thermistek ZHT-630 Heat Alarm/Detector

MAblewicz opened the pull request.

Commit:
Merge remote-tracking branch 'upstream/master'

Commit:
Merge remote-tracking branch 'upstream/master'

Commit:
DEVC-580 Create generic basic heat sensor DTH for FireAngel ZHT-630 Heat
Detector/Alarm

create new z-wave-basic-heat-alarm.groovy for temperature alarm capability
add FireAngel ZHT-630 support
create read.me

Commit:
DEVC-580
Copyright date correction
Indent changes to tabs
Device name correction

Commit:
DEVC-580 Create generic basic heat sensor DTH for FireAngel Thermistek ZHT-630 Heat Alarm/Detector
Event value for smoke clear fixed

greens commented on devicetypes/smartthings/zwave-basic-heat-alarm.src/.st-ignore:
```diff
@@ -0,0 +1,2 @@
+.st-ignore
+README.md
```
This file shouldn't be tracked.

MAblewicz replied:
Sure. I will remove it.

greens commented on devicetypes/smartthings/zwave-basic-heat-alarm.src/zwave-basic-heat-alarm.groovy:
```diff
@@ -0,0 +1,165 @@
+		capability "Health Check"
+
+		//zw:S type:0701 mfr:026F prod:0001 model:0002 ver:1.07 zwv:4.24 lib:03 cc:5E,86,72,5A,73,80,71,85,59,84 role:06 ff:8C01 ui:8C01
+		fingerprint mfr: "026F ", prod: "0001", model: "0002", deviceJoinName: "FireAngel Thermistek ZHT-630 Heat Alarm/Detector"
```
Since this is a generic device type, can we also add a generic device fingerprint based on device type and command classes (deviceId and inClusters)?

MAblewicz replied:
The may be one problem with that. Same classes are used by FireAngel Smoke sensor. Only diffrence is model number 0001 instead of 0002.

greens replied:
bummer

greens commented on devicetypes/smartthings/zwave-basic-heat-alarm.src/zwave-basic-heat-alarm.groovy:
```diff
@@ -0,0 +1,165 @@
+	sendEvent(name: "checkInterval", value: 2 * 60 * 60 + 2 * 60, displayed: false, data: [protocol: "zwave", hubHardwareId: device.hub.hardwareID])
+
+	def cmds = []
+	createHeatEvents("heatClear", cmds)
```
If this only ever returns one event, you can make these lines more concise.

MAblewicz replied:
Thanks. I will correct that.

greens commented on devicetypes/smartthings/zwave-basic-heat-alarm.src/zwave-basic-heat-alarm.groovy:
```diff
@@ -0,0 +1,165 @@
+		switch (cmd.event) {
+			case 0x00:
+			case 0xFE:
+				result = createHeatEvents("heatClear")
```
I have a similar comment here as I did for the smoke detector changes.

Namely that 0x00 should clear both "heat" and "tested" and that 0xFE shouldn't do anything.

greens replied:
Actually, nevermind. I realize now that all of this is only for the single "heat" variable that can be either "detected", "clear", or "test" and that "heatClear" and "testClear" are functionally identical.

MAblewicz replied:
OK. No changes required then.

greens reviewed (changes requested):
A couple changes.

Commit:
Devc 580 Create generic basic heat sensor DTH for FireAngel Thermistek ZHT-630 Heat Alarm/Detector
Post Review Fixes
Add correct health check interval
Unnecessary files removal
installed function fixed
Readme.MD corrected

Commit:
DEV-580 Create generic basic heat sensor DTH for FireAngel Thermistek ZHT-630 Heat Alarm/Detector
Device Name fix

dkirker commented on devicetypes/smartthings/zwave-basic-heat-alarm.src/zwave-basic-heat-alarm.groovy:
```diff
@@ -0,0 +1,164 @@
+				result = createHeatEvents("heat")
+				break
+			case 0x07:
+				result = createHeatEvents("tested")
```
Just an FYI, the Temperature Alarm capability doesn't define "tested" as a value of the attribute "temperatureAlarm", so you *may* run in to some trouble. (Secret note: The OCF mapping just does a pass-through, so as long as any sort of event marshal doesn't block the event, it should be passed through. If we want a "tested" value, we should look at getting it added.)

greens replied:
if it comes up, we can fix it I suppose.

Missed this comment. Sorry.
